custom/ewsghana/reports/specific_reports/stock_status_report.py

```diff
@@ -0,0 +1,324 @@
+                                 "with_stock": with_stock,
+                                 "without_stock": without_stock,
+                                 "without_data": without_data})
+        return rows
+
+    @property
+    def chart_config(self):
+        return {
+            'label_color': {
+                "Stocked out": "#a30808",
+                "Not Stocked out": "#7aaa7a",
+                "No Stock Data": "#efde7f"
+            },
+            'div': "product_availability_summary_plot_placeholder",
+            'legenddiv': "product_availability_summary_legend",
+            'xaxistitle': "Products",
+            'yaxistitle': "Facilities",
+        }
+
+    @property
+    def charts(self):
+        product_availability = self.rows
+        if product_availability:
+            def convert_product_data_to_stack_chart(rows, chart_config):
+                ret_data = []
+                for k in ['Stocked out', 'Not Stocked out', 'No Stock Data']:
+                    datalist = []
+                    for row in rows:
+                        if k == 'No Stock Data':
+                            datalist.append([row['product_code'], row['without_data']])
+                        elif k == 'Stocked out':
+                            datalist.append([row['product_code'], row['without_stock']])
+                        elif k == 'Not Stocked out':
+                            datalist.append([row['product_code'], row['with_stock']])
+                    ret_data.append({'color': chart_config['label_color'][k], 'label': k, 'data': datalist})
+                return ret_data
+
+            chart = MultiBarChart('', x_axis=Axis('Products'), y_axis=Axis(''))
+            chart.rotateLabels = -45
+            chart.marginBottom = 120
+            chart.stacked = False
+            for row in convert_product_data_to_stack_chart(product_availability, self.chart_config):
+                chart.add_dataset(row['label'], [
+                    {'x': r[0], 'y': r[1]}
+                    for r in sorted(row['data'], key=lambda x: x[0])], color=row['color']
+                )
+            return [chart]
+        return []
+
+
+class MonthOfStockProduct(EWSData):
+
+    slug = 'mos_product'
+    title = 'Current MOS by Product'
+    show_chart = False
+    show_table = True
+    use_datatables = True
+
+    @property
+    def headers(self):
+        headers = DataTablesHeader(*[DataTablesColumn('Location')])
+
+        for product in get_products(self.config['program'],
+                                    self.config['products'],
+                                    self.config['domain']).order_by('code'):
+            headers.add_column(DataTablesColumn(product.code))
+
+        return headers
+
+    @property
+    def rows(self):
+        products = get_products(self.config['program'],
+                                self.config['products'],
+                                self.config['domain']).order_by('code')
```

Comment:
Should probably avoid building the products list twice for this report, though they're both pretty basic SQL queries so it's not too bad.